runtime/src/snapshot_bank_utils.rs

```diff
@@ -400,9 +400,27 @@ pub fn bank_from_snapshot_dir(
     );
     info!("{}", measure_rebuild_bank);
 
-    // Skip bank.verify_snapshot_bank.  Subsequent snapshot requests/accounts hash verification requests
-    // will calculate and check the accounts hash, so we will still have safety/correctness there.
-    bank.set_initial_accounts_hash_verification_completed();
+    if bank
+        .feature_set
+        .is_active(&feature_set::accounts_lt_hash::id())
+    {
+        // Skip bank.verify_snapshot_bank.  Subsequent snapshot requests/accounts hash verification requests
+        // will calculate and check the accounts hash, so we will still have safety/correctness there.
+        bank.set_initial_accounts_hash_verification_completed();
```

Comment:
nit: perhaps we could add a log here to indicate that we skip hash verification.